command/peering/export/export.go

```diff
@@ -0,0 +1,187 @@
+	peerNames        string
+}
+
+func (c *cmd) init() {
+	//This function defines the flags
+
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	c.flags.StringVar(&c.serviceName, "service", "", "(Required) Specify the name of the service you want to export.")
+	c.flags.StringVar(&c.serviceNamespace, "serviceNamespace", "", "(Optional) Specify the Namespace of the service, you want to export. Enterprise only feature")
+	c.flags.StringVar(&c.peerNames, "peers", "", "(Required) A list of peers to export the service to, formatted as a comma-separated list.")
+
+	c.http = &flags.HTTPFlags{}
+	flags.Merge(c.flags, c.http.ClientFlags())
+	flags.Merge(c.flags, c.http.PartitionFlag())
+	c.help = flags.Usage(help, c.flags)
+}
+
+func (c *cmd) Run(args []string) int {
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+
+	if c.serviceName == "" {
+		c.UI.Error("Missing the required -service flag")
+		return 1
+	}
+
+	if c.peerNames == "" {
+		c.UI.Error("Missing the required -peers flag")
+		return 1
+	}
+
+	peerNames := strings.Split(c.peerNames, ",")
+
+	for _, peerName := range peerNames {
+		if peerName == "" {
+			c.UI.Error(fmt.Sprintf("Invalid peer %q", peerName))
+			return 1
+		}
+	}
+
+	client, err := c.http.APIClient()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error connect to Consul agent: %s", err))
+		return 1
+	}
+
+	entry, _, err := client.ConfigEntries().Get("exported-services", "default", nil)
+	if err != nil && !strings.Contains(err.Error(), agent.ConfigEntryNotFoundErr) {
+		c.UI.Error(fmt.Sprintf("Error reading config entry %s/%s: %v", "exported-services", "default", err))
+		return 1
+	}
+	if entry == nil {
+		cfg := api.ExportedServicesConfigEntry{
+			Name: "default",
+			Services: []api.ExportedService{
+				{
+					Name:      c.serviceName,
+					Namespace: c.serviceNamespace,
+					Consumers: buildConsumersFromPeerNames(peerNames),
+				},
+			},
+		}
+
+		_, _, err = client.ConfigEntries().Set(&cfg, &api.WriteOptions{})
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error creating config entry: %s", err))
+			return 1
+		}
+
+	} else {
+
+		cfg, ok := entry.(*api.ExportedServicesConfigEntry)
+		if !ok {
+			c.UI.Error(fmt.Sprintf("Existing config entry has incorrect type: %t", entry))
+			return 1
+		}
+
+		serviceExists := false
+
+		for i, service := range cfg.Services {
+
+			if service.Name == c.serviceName && service.Namespace == c.serviceNamespace {
+
+				serviceExists = true
+				for _, peerName := range peerNames {
+					peerExists := false
+
+					for _, consumer := range service.Consumers {
+						if consumer.Peer == peerName {
+							peerExists = true
+							break
+						}
+					}
+
+					if !peerExists {
+						cfg.Services[i].Consumers = append(cfg.Services[i].Consumers, api.ServiceConsumer{Peer: peerName})
+					}
+				}
+			}
+		}
+
+		if !serviceExists {
+			cfg.Services = append(cfg.Services, api.ExportedService{
+				Name:      c.serviceName,
+				Namespace: c.serviceNamespace,
```

Comment:
Notably, we can assign a namespace here even when using OSS. Discussed in the issue w/ @DanStough that we will want to update the config-entry validation on the server to return an error such as `enterprise-only feature`.